### PR TITLE
[IA-4033] Standardize Interactive Analysis mixpanel metrics.

### PR DESCRIPTION
**Pull request by n8landolt**
https://broadworkbench.atlassian.net/wiki/spaces/IA/pages/2711715862/Analysis+Mixpanel+Metrics

The new source of truth for Interactive Analysis front end mixpanel metrics.

There have been a few deprecated metrics.
notebook:Launch - is now cloudEnvironmentLaunch.
analysis:azureJupyterLabCreate - see cloudEnvironmen:create and filter by cloudProvider:'AZURE'
analysis:Launch - deprecated in favor of cloudEnvironment:launch
cloudEnvironment:create - deprecated in favor of cloudEnvironment:create

### CODE REVIEW

File: src/pages/workspaces/workspace/analysis/Analyses.ts [hunk 1]
```diff
@@ -264,7 +264,7 @@ const activeFileTransferMessage = div({
 ])
 
 export type DisplayAnalysisFile = AnalysisFile & {
-  application: ToolLabel
+  application: ToolLabel // TODO: Shouold be renamed tool
```
Comment (LizBaldo) [inline src/pages/workspaces/workspace/analysis/Analyses.ts]: Is this TODO too big of a refactoring to do as part of this PR?

File: src/pages/workspaces/workspace/analysis/AnalysisLauncher.js [hunk 2]
```diff
@@ -567,6 +576,7 @@ const AnalysisEditorFrame = ({
   const [busy, setBusy] = useState(false)
   const [analysisSetupComplete, setAnalysisSetupComplete] = useState(false)
   const cookieReady = useStore(cookieReadyStore)
+  const cloudPlatform = !!googleProject ? cloudProviderTypes.GCP : cloudProviderTypes.AZURE
```
Comment (LizBaldo) [inline src/pages/workspaces/workspace/analysis/AnalysisLauncher.js]: naive js question on my part, but why does this constant has to be defined the same way again? is there a DRY way to set this globally?

File: src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js [hunk 3]
```diff
@@ -191,13 +194,31 @@ export const AzureComputeModalBase = ({
     )
   }
 
+  const sendCloudEnvironmentMetrics = () => {
+    const metricsEvent = Utils.cond(
+      [(viewMode === 'deleteEnvironment'), () => 'cloudEnvironmentDelete'],
+      //[(!!existingRuntime), () => 'cloudEnvironmentUpdate'], TODO: When update is available, include in metrics
```
Comment (LizBaldo) [inline src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js]: Would it make sense to create a ticket in the backlog summarizing the updates needed and link here?

File: src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js [hunk 4]
```diff
@@ -191,13 +194,31 @@ export const AzureComputeModalBase = ({
...
+    )
+
+    //TODO: When update is available include existingRuntime in metrics.
+    Ajax().Metrics.captureEvent(Events[metricsEvent], {
+      ...extractWorkspaceDetails(workspace),
+      ..._.mapKeys(key => `desiredRuntime_${key}`, computeConfig),
+      desiredRuntime_region: computeConfig.region,
+      desiredRuntime_machineType: computeConfig.machineType,
+      desiredPersistentDisk_size: computeConfig.diskSize,
+      desiredPersistentDisk_type: 'Standard', //TODO: Hard coded - we currently don't send disk type.
```
Comment (LizBaldo) [inline src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js]: Would it make more sense to omit it, or send a `NaN` instead?

**Review by LizBaldo — APPROVED** [review]
LGTM, I am not very familiar with this part of the code or the language yet, but the changes fit with the ticket and description provided :) 
I do have a couple of suggestions to keep track of the TODOs and replace a hardcoded value, lmk what you think!

File: src/pages/workspaces/workspace/analysis/AnalysisLauncher.js [hunk 5]
```diff
@@ -467,6 +469,7 @@ const AnalysisPreviewFrame = ({ analysisName, toolLabel, workspace: { workspace:
   const [busy, setBusy] = useState(false)
   const [preview, setPreview] = useState()
   const frame = useRef()
+  const cloudPlatform = !!googleProject ? cloudProviderTypes.GCP : cloudProviderTypes.AZURE
```
Comment (jdcanas) [inline src/pages/workspaces/workspace/analysis/AnalysisLauncher.js]: can we use `getCloudProviderFromWorkspace` here ?

File: src/pages/workspaces/workspace/analysis/AnalysisLauncher.js (hunk 2 above)
Comment (jdcanas) [reply src/pages/workspaces/workspace/analysis/AnalysisLauncher.js]: its in a different functional component, so even if its in the same file they have different scopes. Realistically, the workspace store should have the `cloudPlatform` information, but short of that we should use the function `getCloudProviderFromWorkspace` in my opinion

File: src/pages/workspaces/workspace/analysis/AnalysisLauncher.js [hunk 6]
```diff
@@ -640,6 +650,7 @@ const WelderDisabledNotebookEditorFrame = ({
   const [busy, setBusy] = useState(false)
   const [localized, setLocalized] = useState(false)
   const cookieReady = useStore(cookieReadyStore)
+  const cloudPlatform = !!googleProject ? cloudProviderTypes.GCP : cloudProviderTypes.AZURE
```
Comment (jdcanas) [inline src/pages/workspaces/workspace/analysis/AnalysisLauncher.js]: `getCloudProviderFromWorkspace`

**Review by jdcanas — APPROVED** [review]
few small nits

File: src/pages/workspaces/workspace/analysis/Analyses.ts (hunk 1 above)
Comment (n8landolt) [reply src/pages/workspaces/workspace/analysis/Analyses.ts]: Eh I'm just going to remove the comment, not sure that it's really necessary.

File: src/pages/workspaces/workspace/analysis/AnalysisLauncher.js (hunk 2 above)
Comment (n8landolt) [reply src/pages/workspaces/workspace/analysis/AnalysisLauncher.js]: Yeah. We could use some updated data models.

File: src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js (hunk 3 above)
Comment (n8landolt) [reply src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js]: Yes, I'll get on that - I'll make the ticket strictly for mixpanel metrics and we can link it to the update feature ticket, which I'm unsure if it exists

File: src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js (hunk 4 above)
Comment (n8landolt) [reply src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js]: My thinking is this - we default users to the standard disk type so we can send this with confidence, until we allow choosing a disk type.
I'm not opposed to omit it until the user has actual choices, though.

File: src/pages/workspaces/workspace/analysis/AnalysisLauncher.js [hunk 7]
```diff
@@ -462,26 +464,28 @@ const PreviewHeader = ({
...
 
 // This component is responsible for rendering the html preview of the analysis file
-const AnalysisPreviewFrame = ({ analysisName, toolLabel, workspace: { workspace: { workspaceId, googleProject, bucketName } }, onRequesterPaysError, styles }) => {
+const AnalysisPreviewFrame = ({ analysisName, toolLabel, workspace, onRequesterPaysError, styles }) => {
   const signal = useCancellation()
   const [busy, setBusy] = useState(false)
   const [preview, setPreview] = useState()
   const frame = useRef()
+  const cloudPlatform = getCloudProviderFromWorkspace(workspace)
+  const { workspace: { workspaceId, googleProject, bucketName } } = workspace
```
Comment (jdcanas) [inline src/pages/workspaces/workspace/analysis/AnalysisLauncher.js:473]: thanks. A pit pedantic, but I find using workspace likes this makes it easier when it comes time to type the props!

**Pull request merged**